### PR TITLE
Improve default inputs

### PR DESCRIPTION
**Pull request by lennym**
If no values are provided then it attempts to infer from the field name what type of input is required, and loads a corresponding value from `hof-util-testdata`.

For select boxes and radio groups, when no value is available then it picks a random (for selects ignoring option 0) option.

For checkboxes, if no value if defined then all checkboxes will be selected.

### CODE REVIEW

File: lib/autofill.js [hunk 1]
```diff
@@ -2,93 +2,118 @@
 
 const Promise = require('bluebird');
 const url = require('url');
+const Inputs = require('./inputs');
```
Comment (joefitter) [inline lib/autofill.js:5]: If I were a pedant I would say that `Inputs` exports a function not a class so the variable name shouldn't be capitalised. I don't actually care that much though

File: lib/autofill.js [hunk 2]
```diff
@@ -2,93 +2,118 @@
...
+    }
   }
 
   function completeStep(path) {
     return browser
       .elements('input')
       .then(fields => {
         debug(`Found ${fields.value.length} <input> elements`);
-        const fillers = fields.value.map(field => {
+        return Promise.map(fields.value, field => {
```
Comment (joefitter) [inline lib/autofill.js:80]: what is this trickery! `Promise.map` isn't a 'thing' with native promises... hmm... Bluebird eh?

File: lib/autofill.js [hunk 3]
```diff
@@ -2,93 +2,118 @@
...
                   } else if (type.value === 'text') {
                     return completeTextField(field.ELEMENT, name.value);
                   } else {
                     debug(`Ignoring field of type ${type.value}`);
                   }
                 });
             });
-        });
-        return Promise.all(fillers);
+        }, {concurrency: 1});
```
Comment (joefitter) [inline lib/autofill.js:96]: many disappoint :(

**Review by joefitter — APPROVED** [review]
✋

File: lib/autofill.js (hunk 2 above)
Comment (lennym) [reply lib/autofill.js:80]: Yeah, webdriverio has some nasty internal state somewhere that causes sporadic bugs if you try to complete fields in parallel so they need to be done one at a time. I'd rather have been able to use native `Promise.all` too.

**Pull request merged**